Updates to getting started instructions

jeremiak opened the pull request:
@juliaelman and I were pairing on getting her local environment set up and the instructions were a bit out of date.

Mostly, this added some instructions on cloning the repository so that it fits with a typical Go project structure.

I also changed references from "Angular" to "React" so that all the information is accurate for the rewrite branch.

juliaelman commented on README.md:
```diff
@@ -15,10 +15,16 @@ Learn more about [cloud.gov](https://cloud.gov).
 
-- `AngularJS` for the frontend. [![JS Code Coverage Status](http://codecov.io/github/18F/cg-deck/coverage.svg?branch=master)](http://codecov.io/github/18F/cg-deck?branch=master)
-
+- `Node` (v.4.2.3) and `React` for the frontend. [![JS Code Coverage Status](https://coveralls.io/repos/18F/cg-deck/badge.svg?branch=staging-alpha&service=github)](https://coveralls.io/repos/18F/cg-deck/badge.svg?branch=staging-alpha&service=github)
```
I'd recommend putting React on it's own line and reference the version number. Probably something important to understand when debugging and also to reference during installation.

juliaelman commented on README.md:
```diff
@@ -15,10 +15,16 @@ Learn more about [cloud.gov](https://cloud.gov).
 
 ## Setup
+### Cloning the repository
+If you are unfamiliar with `Go` project directory structure, you want the code in this repository to be in something like `<your-code-directory>/cg-deck-ws/src/github.com/18f/cg-deck`. You can use that exact pattern by cloning the repository with:
```
Maybe make `project directory structure` hyperlink to: https://golang.org/doc/code.html#Workspaces

jeremiak replied:
Great idea

juliaelman commented on README.md:
````diff
@@ -15,10 +15,16 @@ Learn more about [cloud.gov](https://cloud.gov).
+
+```
+git clone git@github.com:18F/cg-deck.git cg-deck-ws/src/github.com/18F/cg-deck
+```
````
Since (I think) `ws` stands for `workspace`, we might want to mention this and why the clone adds that in. Not 100% sure it HAS to be explicit, but might be helpful for folks not familiar with the Go file structure requirements.

juliaelman commented on README.md:
````diff
@@ -15,10 +15,16 @@ Learn more about [cloud.gov](https://cloud.gov).
+git clone git@github.com:18F/cg-deck.git cg-deck-ws/src/github.com/18F/cg-deck
+```
+
 ### Create a Client with UAAC
````
@jeremiak mentioned to me that this may or may not be necessary for local dev. If so, should we state either why it is helpful and if it is a requirement or not?

jeremiak replied:
I think this is still necessary, I just gave you my generated client tokens when we were getting your environment setup. In fact, if you could test out these instructions to make sure they work that would be really valuable.

juliaelman commented on README.md:
```diff
@@ -34,28 +40,24 @@ uaac client add <your-client-id> \
 
 ### Set the environment variables
-If you are testing locally, export these variables. If you are deploying to cloud foundry, modify the manifest.yml
+If you are testing locally, export these variables. If you are deploying to cloud foundry, modify the `manifest.yml`
```
Should we add advice on best practices for exporting the environment variables?

juliaelman replied:
Or maybe we could add an example file for folks to cp and set?

juliaelman commented on README.md:
```diff
@@ -34,28 +40,24 @@ uaac client add <your-client-id> \
 
 ## Front end
-Install front end dependencies, including the `cloudgov-style` library
+Install front end dependencies:
```
Add note about where in the dir structure to run this cmd from. e.g. `/path/to/folder/cg-deck-ws/src/github.com/18F/cg-deck`

juliaelman commented:
Looks good and thank you for taking the time to do this! Added a few notes that might help clarify things a bit further. I was able to get up and running with only a few road bumps (which I was able to figure out!). ^5

jeremiak commented:
Thanks for the feedback @juliaelman, I think I addressed most of your points. Great idea about a sample file of environment variables. Ready for your review!

juliaelman commented:
LGTM - thanks @jeremiak!
